src/hotspot/share/gc/shenandoah/shenandoahBarrierSet.inline.hpp

```diff
@@ -422,17 +422,58 @@ void ShenandoahBarrierSet::arraycopy_barrier(T* src, T* dst, size_t count) {
       // Note that we can't do the arraycopy marking using the 'src' array when
       // SATB mode is enabled (so we can't do this as part of the iteration for
       // evacuation or update references).
-      arraycopy_marking(src, dst, count);
+      arraycopy_marking(src, dst, count, true);
     }
   }
 }
 
 template <class T>
-void ShenandoahBarrierSet::arraycopy_marking(T* src, T* dst, size_t count) {
+void ShenandoahBarrierSet::arraycopy_marking(T* src, T* dst, size_t count, bool is_old) {
```

Comment:
Oh, this `is_old` is not about the object, it is about the marking! Maybe name it `is_old_marking`?